backend/__init__.py

```diff
@@ -22,6 +22,11 @@ def create_app(config=Config):
     login.init_app(app)
     Migrate(app, db)
 
+    def internal_error(error):
+        db.session.rollback()
```

Comment:
This function needs to be defined inside here?

Reply:
Not really, but it's a two-liner and only ever needed here.